packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -410,106 +570,655 @@ class _CupertinoAlertActionSection extends StatelessWidget {
+        if (_isSingleButtonRow(width)) {
+          // The first 2 buttons fit side-by-side. Display them horizontally.
+          return _computeMinIntrinsicHeightSideBySide(width);
+        } else {
+          // The first 2 buttons do not fit side-by-side. Display them stacked.
+          // The minimum height for 2 buttons when stacked is the minimum height
+          // of both buttons + dividers (no scrolling for 2 buttons).
+          return _computeMinIntrinsicHeightForTwoStackedButtons(width);
+        }
+      } else {
+        // 3+ buttons are always stacked. The minimum height when stacked is
+        // 1.5 buttons tall.
+        return _computeMinIntrinsicHeightStacked(width);
+      }
+    }
+  }
+
+  // The minimum height for a single row of buttons is the larger of the buttons'
+  // min intrinsic heights + the width of a divider
+  double _computeMinIntrinsicHeightSideBySide(double width) {
+    assert(childCount <= 2);
+
+    if (childCount == 1) {
+      return firstChild.computeMinIntrinsicHeight(width) + dividerThickness;
+    } else {
+      final double perButtonWidth = (width - dividerThickness) / 2.0;
+      return math.max(
+        firstChild.computeMinIntrinsicHeight(perButtonWidth) + dividerThickness,
+        lastChild.computeMinIntrinsicHeight(perButtonWidth) + dividerThickness,
+      );
+    }
+  }
+
+  // The minimum height for 2 stacked buttons is the height of both
+  // buttons + 2 dividers.
+  double _computeMinIntrinsicHeightForTwoStackedButtons(double width) {
+    assert(childCount == 2);
+
+    return (2 * dividerThickness)
+        + firstChild.computeMinIntrinsicHeight(width)
+        + lastChild.computeMinIntrinsicHeight(width);
+  }
+
+  // The minimum height for 3+ stacked buttons is the height of the 1st button
+  // + 50% the height of the 2nd button + 2 dividers.
+  double _computeMinIntrinsicHeightStacked(double width) {
+    assert(childCount >= 3);
+
+    final List<RenderBox> children = getChildrenAsList();
+    return (2 * dividerThickness)
+        + children[0].computeMinIntrinsicHeight(width)
+        + (0.5 * children[1].computeMinIntrinsicHeight(width));
+  }
+
+  @override
+  double computeMaxIntrinsicHeight(double width) {
+    if (childCount == 0) {
+      // No buttons. Zero height.
+      return 0.0;
+    } else if (childCount == 1) {
+      // One button. Our max intrinsic height is equal to the button's.
+      return firstChild.computeMaxIntrinsicHeight(width) + dividerThickness;
+    } else if (childCount == 2) {
+      // Two buttons...
+      if (_isSingleButtonRow(width)) {
+        // The 2 buttons fit side by side so our max intrinsic height is equal
+        // to the taller of the 2 buttons.
+        final double perButtonWidth = (width - dividerThickness) / 2.0;
+        return math.max(
+          firstChild.computeMaxIntrinsicHeight(perButtonWidth),
+          lastChild.computeMaxIntrinsicHeight(perButtonWidth),
+        ) + dividerThickness;
```

Comment:
this math is oddly different than your similar case before (where you had the dividerThickness inside the max)

Reply:
(outside is better)